packages/headless/src/features/question-answering/question-answering-analytics-actions.ts

```diff
@@ -1,10 +1,31 @@
+import {CoveoSearchPageClient} from 'coveo.analytics';
 import {validatePayload} from '../../utils/validate-payload';
 import {AnalyticsType, makeAnalyticsAction} from '../analytics/analytics-utils';
 import {
   documentIdentifierPayloadDefinition,
   QuestionAnsweringDocumentIdActionCreatorPayload,
 } from './question-answering-document-id';
 
+type AnalyticsSmartSnippetFeedbackReason = Parameters<
+  CoveoSearchPageClient['logSmartSnippetFeedbackReason']
+>[0];
```

Comment:
Had lots of trouble with this. The `SmartSnippetFeedbackReason` enum isn't exported from `coveo.analytics`. I'll open a PR to export it in `coveo.analytics`.

Reply:
I think another solution would have been to return the string values `as const`.

I think TypeScript would have correctly detected that since they are constant, they match the enum.

Reply:
I already tried something similar. Unfortunately, it doesn't. Here's what it shows with `as const`:
![image](https://user-images.githubusercontent.com/54454747/159352910-13b7e796-c6f6-43df-88fd-4bd399909c97.png)